src/SchemaRegistry.ts

```diff
@@ -24,8 +26,12 @@ const DEFAULT_OPTS = {
   separator: DEFAULT_SEPERATOR,
 }
 
+type RegistryId = number
+
 export default class SchemaRegistry {
   private api: SchemaRegistryAPIClient
+  private cacheMissRequests: Map<RegistryId, Promise<Response>> = new Map()
```

Comment:
Since we're only going to do lookups, there's no need for this to be a `Map`. It also seem to complicate things with TypeScript, forcing usage of `non-null-assertion`

```suggestion
  private cacheMissRequests: { [key: number]: Promise<Response> } = {}
```